Runtime/Scripts/TileHandler.cs

```diff
@@ -637,40 +589,39 @@ private void RemoveOutOfViewTiles()
                 {
                     continue;
                 }
-                if (layer.gameObject.activeSelf == false) { continue; }
-                if (layer.isEnabled == false)
-                {
-                    continue;
-                }
+
+                if (layer.gameObject.activeSelf == false) continue;
+                if (layer.isEnabled == false) continue;
+                
                 int tilesizeIndex = tileSizes.IndexOf(layer.tileSize);
-                neededTiles = tileDistances[tilesizeIndex];
-                neededTileKeys.Clear();
-                neededTileKeys.Capacity = neededTiles.Count;
-                foreach (var neededTile in neededTiles)
-                {
-                    //tileKey.x = neededTile.x;
-                    //tileKey.y = neededTile.y;
-                    neededTileKeys.Add(new Vector2Int(neededTile.x, neededTile.y));
-                }
-                //activeTiles = layer.tiles.Keys.ToArray();
-                //activeTiles = new List<Vector2Int>(layer.tiles.Keys);
+                var neededTiles = tileDistances[tilesizeIndex];
+
+                var neededTileKeys = new HashSet<Vector2Int>(
```

Comment:
Performance improvement: hashset lookups are O(1) where List lookups (using Contains) are O(n). This code is also simplified and will no longer use class-wide state for its internals